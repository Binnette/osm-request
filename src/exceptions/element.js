export function WrongElementTypeException(message) {
  this.message = message;
  this.name = 'WrongElementTypeException';
}
