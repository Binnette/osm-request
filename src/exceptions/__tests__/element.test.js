import { WrongElementTypeException } from '../element';

describe('Element exceptions', () => {
  describe('WrongElementTypeException', () => {
    it('Should be well formatted', () => {
      const message = 'This is a test';
      const myFunction = () => {
        throw new WrongElementTypeException(message);
      };

      expect(myFunction).toThrow(WrongElementTypeException);
      expect(myFunction).toThrow(message);
      expect(myFunction).toThrowErrorMatchingSnapshot();
    });
  });
});
