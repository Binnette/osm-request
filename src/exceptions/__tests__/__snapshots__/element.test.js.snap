// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Element exceptions WrongElementTypeException Should be well formatted 1`] = `"This is a test"`;
